lib/llvmopencl/FixUnreachable.cc

```diff
@@ -0,0 +1,146 @@
+// FixUnreachable is an LLVM pass to fix unreachable instructions to defined
+// behavior (mark external variable + return)
+//
+// Copyright (c) 2025 Michal Babej / Intel Finland Oy
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "CompilerWarnings.h"
+IGNORE_COMPILER_WARNING("-Wmaybe-uninitialized")
+#include <llvm/ADT/Twine.h>
+POP_COMPILER_DIAGS
+IGNORE_COMPILER_WARNING("-Wunused-parameter")
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/IR/Instructions.h>
+
+#include "FixUnreachable.h"
+#include "LLVMUtils.h"
+#include "WorkitemHandlerChooser.h"
+POP_COMPILER_DIAGS
+
+#include <iostream>
+#include <map>
+#include <set>
+
+#define PASS_NAME "fix-unreachable"
+#define PASS_CLASS pocl::FixUnreachable
+#define PASS_DESC "Fix unreachable instruction uses to defined behavior"
+
+// #define DEBUG_FIX_UNREACHABLE
+
+namespace pocl {
+
+using namespace llvm;
+
+static bool fixUnreachable(Function &F) {
```

Comment:
"Fix" is not obvious here, please add a comment what, why and how is "fixed".

Reply:
renamed & added comment on the Pass in the file's header